apollo-router/src/plugin/mod.rs

```diff
@@ -312,6 +320,16 @@ where
     }
 }
 
+impl dyn DynPlugin {
+    pub(crate) fn downcast_ref<T: 'static>(&self) -> Option<&T> {
+        if TypeId::of::<T>() == self.type_id() {
+            unsafe { Some(&*(self as *const dyn DynPlugin as *const T)) }
```

Comment:
```suggestion
            Some(unsafe { &*(self as *const _ as *const T) })
```

Reply:
This line was copy-pasted from the standard library, but I don’t mind making that change if you prefer. I can also add a comment with the source URL

Reply:
I also tend to prefer not relying on inference too much when casting raw pointers, it’s easy to end up accidentally casting from `&&Foo` instead of `&Foo`

Reply:
Good points. How about:
```
Some(unsafe { &*(self as *const dyn DynPlugin as *const T) })
```
?

Reply:
Done!

Reply:
++ on the source url that would be nice ^^